src/routes/forms.go

```diff
@@ -0,0 +1,250 @@
+			resData = append(resData, dtos.CreateUpdateGetFormResponseDto{
+				Id:          f.Id,
+				Content:     f.Content,
+				OwnerId:     f.OwnerId,
+				CreatedById: f.CreatedById,
+				UpdatedById: f.UpdatedById,
+				Status:      string(f.Status),
+				CreatedAt:   f.CreatedAt.String(),
+				UpdatedAt:   f.UpdatedAt.String(),
+			})
+		}
+
+		var res = dtos.ResponseDto{
+			Message: "forms fetched successfully",
+			Data:    resData,
+		}
+
+		ctx.JSON(http.StatusOK, res)
+	})
+
+	forms.GET("/:id", func(ctx *gin.Context) {
+		var form models.Form
+		if err := db.NewSelect().Model(&form).Where("id = ?", ctx.Param("id")).Scan(ctx); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "something went wrong",
+				Error: &dtos.ErrorResponse{
+					Message: "error fetching form",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		var formMetaData models.FormMetaData
+		if err := db.NewSelect().Model(&formMetaData).Where("form_id = ?", ctx.Param("id")).Scan(ctx); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "something went wrong",
+				Error: &dtos.ErrorResponse{
+					Message: "error fetching form meta data",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
+
+		var resData = dtos.GetFormDetailResponseDto{
+			Id:          form.Id,
+			OwnerId:     form.OwnerId,
+			Status:      string(form.Status),
+			CreatedById: form.CreatedById,
+			UpdatedById: form.UpdatedById,
+			CreatedAt:   form.CreatedAt.String(),
+			UpdatedAt:   form.UpdatedAt.String(),
+			Content:     form.Content,
+			Meta: dtos.GetFormMetaDataResponseDto{
+				Id:                               formMetaData.Id,
+				FormId:                           formMetaData.FormId,
+				IsDeleted:                        formMetaData.IsDeleted,
+				AccepctingResponses:              formMetaData.AccepctingResponses,
+				AllowGuestResponses:              formMetaData.AllowGuestResponses,
+				AllowMultipleRepsonses:           formMetaData.AllowMultipleRepsonses,
+				SendConfirmationEmailToRespondee: formMetaData.SendConfirmationEmailToRespondee,
+				SendSubmissionEmailToOwner:       formMetaData.SendSubmissionEmailToOwner,
+				ValidTill:                        formMetaData.ValidTill,
+				UpdatedById:                      formMetaData.UpdatedById,
+				UpdatedAt:                        formMetaData.UpdatedAt,
+			},
+		}
+
+		var res = dtos.ResponseDto{
+			Message: "form fetched successfully",
+			Data:    resData,
+		}
+
+		ctx.JSON(http.StatusOK, res)
+	})
+
+	forms.PATCH("/:id", func(ctx *gin.Context) {
+		var requestBody dtos.CreateUpdateFormRequestDto
+		if err := ctx.ShouldBindJSON(&requestBody); err != nil {
+			errObj := dtos.ResponseDto{
+				Message: "invalid request",
+				Error: &dtos.ErrorResponse{
+					Message: "invalid request body",
+					Detail:  err.Error(),
+				},
+			}
+			ctx.JSON(http.StatusBadRequest, errObj)
+			return
+		}
```

Comment:
NIT: can we move validation to a separate file?

Reply:
https://github.com/Real-Dev-Squad/wisee-backend/issues/37